File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int countletter(string texts);
int countword(string text0);
int countsentence(string text1);
int formula(int l, int w, int s);

int main(void)
{
    // Initialise
    int letters = 0;
    int words = 0;
    int sentences = 0;
    
    // Prompt for string
    string text = get_string("Text: ");
    
    // Prepare for calculate formula
    letters = countletter(text);
    words = countword(text);
    sentences = countsentence(text);

    // Formula count grade
    int formula1 = formula(letters, words, sentences);
    
    // Output
    if (formula1 > 16)
    {
        printf("Grade 16+\n");
    }
    
    if (formula1 < 1)
    {
        printf("Before Grade 1\n");
    }
    
    if (formula1 >= 1 && formula1 <= 16)
    {
        printf("Grade %i\n", formula1);
    }
}




// Count all letters in string
int countletter(string texts)
{
    int lettersum = 0;
    
    for (int i = 0; texts[i] != '\0'; i++)
    {
        bool test0 = isalpha(texts[i]);
        
        if (test0 == true)
        {
            lettersum += 1;
        }
    }
    return lettersum;
}




// Count all words in string
int countword(string text0)
{
    int wordsum = 0;
    
    for (int i = 0; text0[i] != '\0'; i++)
    {
        if (text0[i] == ' ')
        {
            wordsum += 1;
        }
    }
    return wordsum + 1;
}




// Count all sentences in string
int countsentence(string text1)
{
    int stringsum = 0;
    
    for (int i = 0; text1[i] != '\0'; i++)
    {
        if (text1[i] == '.' || text1[i] == '!' || text1[i] == '?')
        {
            stringsum += 1;
        }
    }
    return stringsum;
}




// Count grade
int formula(int l, int w, int s)
{
    double index = 0;
    double L = 0;
    double S = 0;
    
    L = (double)l /  w * 100.0;
    S = (double)s /  w * 100.0;
    
    index = 0.0588 * L - (0.296 * S) - 15.8;
    
    int index1 = round(index);
    
    return index1;
}

File: pset6/readability.py
# Count all letters in string
def count_letter(text):
    sum_letter = 0

    for letter in range(len(text)):
        if text[letter].isalpha():
            sum_letter += 1

    return sum_letter


# Count all words in string
def count_word(text):
    word_list = text.split(" ")

    return len(word_list)


# Count all sentences
def count_sentences(text):
    sum_string = 0

    for word in range(len(text)):
        if text[word] == "." or text[word] == "!" or text[word] == "?":
            sum_string += 1

    return sum_string


# Count grade
def formula(letters, words, sentences):
    l = float(int(letters) / int(words) * 100)
    s = float(int(sentences) / int(words) * 100)

    index = int(round(0.0588 * l - (0.296 * s) - 15.8, 0))

    return index


# Main function
def main():
    # taking input
    text = input("Text: ")

    # Prepare for calculate formula
    letters = count_letter(text)
    words = count_word(text)
    sentences = count_sentences(text)

    # Formula count grade
    grade = formula(letters, words, sentences)

    # Output
    if grade > 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print("Grade ", grade)


# Call main function
main()